Content.Client/UserInterface/HandsGui.cs

```diff
@@ -209,7 +209,7 @@ private void HandKeyBindDown(GUIBoundKeyEventArgs args, string slotName)
 
             if (args.Function == ContentKeyFunctions.MouseMiddle)
             {
-                hands.SendChangeHand(slotName);
+                hands.SetActiveHand(slotName);
```

Comment:
You should not directly edit simulation state from UI like this. As it is this "event" is not correctly tracked by the game state system and will be undone during reconciliation. 

Use `EntityManagerExt.RaisePredictiveEvent()` to fire events into the EventBus and to the server that are correctly replayed during reconciliation. Check the combat mode GUI for an example.

Reply:
Fixed, tell me if it's correct now.